client/api/factory.go

```diff
@@ -0,0 +1,37 @@
+package api
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/lidofinance/dc4bc/client/api/http_api"
+	"github.com/lidofinance/dc4bc/client/config"
+	"os"
+)
+
+type IServerAbstractFactory interface {
+	NewServer(router *gin.Engine, config *config.Config) error
+	Start() error
+	Stop()
+}
+
+type InstanceFactory struct {
+	apiFactory *http_api.RESTApiProvider
+}
+
+var (
+	factoryInstance InstanceFactory
+	done            chan bool
+)
+
+func Run(config *config.Config) {
+	factoryInstance = InstanceFactory{
+		apiFactory: &http_api.RESTApiProvider{},
+	}
+
+	err := factoryInstance.apiFactory.NewServer(config)
+	if err != nil {
+		os.Exit(1)
+	}
+	go factoryInstance.apiFactory.Start()
+
+	<-done
```

Comment:
This global variable chan is not used anywhere. Seems like it's used to just block execution forever? If so, this can be done without global vars